share/eds/cache/accessor_cache.go

```diff
@@ -134,13 +187,27 @@ func (bc *AccessorCache) GetOrLoad(
 
 	// Create a new accessor first to increment the reference count in it, so it cannot get evicted
 	// from the inner lru cache before it is used.
-	ac := newCloser(abs)
+	accessorWithRef, err := newRefCloser(abs)
+	if err != nil {
+		return nil, err
+	}
 	bc.cache.Add(key, abs)
-	return ac, nil
+	return accessorWithRef, nil
 }
 
 // Remove removes the Accessor for a given key from the cache.
 func (bc *AccessorCache) Remove(key shard.Key) error {
+	lk := &bc.stripedLocks[shardKeyToStriped(key)]
+	lk.Lock()
+	accessor, err := bc.get(key)
+	lk.Unlock()
+	if errors.Is(err, errCacheMiss) {
+		// item is not in cache
+		return nil
+	}
+	if err = accessor.close(); err != nil {
+		return err
+	}
```

Comment:
do you need to close accessor here? doesn't it happen in evictFn?

Reply:
it will be closed inside `close()` call

Reply:
I think she means why close it here when it will get close() called in evictFn inside of bc.cache.Remove

Reply:
Yes

Reply:
This close is sync and will yield error to client if it happens. While evict will call close() async without any error returned

Reply:
But doesn't it then get called twice?

Reply:
It will be called second time in evict, but there is no way around it. I've allowed subsequent calls of close() to be noop.